examples/alexnet/alexnet.py

```diff
@@ -0,0 +1,424 @@
+import tensorflow as tf
+from typing import List, Tuple
+
+import ray.array.remote as ra
+
+def load_chunk(tarfile, size=None):
+  """Load a number of images from a single imagenet .tar file.
+
+  This function also converts the image from grayscale to RGB if necessary.
+
+  Args:
+    tarfile (tarfile.TarFile): The archive from which the files get loaded.
+    size (Optional[Tuple[int, int]]): Resize the image to this size if provided.
+
+  Returns:
+    numpy.ndarray: Contains the image data in format [batch, w, h, c]
+  """
+  result = []
+  filenames = []
+  for member in tarfile.getmembers():
+    filename = member.path
+    content = tarfile.extractfile(member)
+    img = Image.open(content)
+    rgbimg = Image.new("RGB", img.size)
+    rgbimg.paste(img)
+    if size != None:
+      rgbimg = rgbimg.resize(size, Image.ANTIALIAS)
+    result.append(np.array(rgbimg).reshape(1, rgbimg.size[0], rgbimg.size[1], 3))
+    filenames.append(filename)
+  return np.concatenate(result), filenames
+
+@ray.remote([str, str, List], [np.ndarray, List])
+def load_tarfile_from_s3(bucket, s3_key, size=[]):
+  """Load an imagenet .tar file.
+
+  Args:
+    bucket (str): Bucket holding the imagenet .tar.
+    s3_key (str): s3 key from which the .tar file is loaded.
+    size (List[int]): Resize the image to this size if size != []; len(size) == 2 required.
+
+  Returns:
+    np.ndarray: The image data (see load_chunk).
+  """
+  s3 = boto3.client("s3")
+  response = s3.get_object(Bucket=bucket, Key=s3_key)
+  output = io.BytesIO()
+  chunk = response["Body"].read(1024 * 8)
+  while chunk:
+    output.write(chunk)
+    chunk = response["Body"].read(1024 * 8)
+  output.seek(0) # go to the beginning of the .tar file
+  tar = tarfile.open(mode="r", fileobj=output)
+  return load_chunk(tar, size=size if size != [] else None)
+
+def load_tarfiles_from_s3(bucket, s3_keys, size=[]):
+  """Load a number of imagenet .tar files.
+
+  Args:
+    bucket (str): Bucket holding the imagenet .tars.
+    s3_keys (List[str]): List of s3 keys from which the .tar files are being loaded.
+    size (List[int]): Resize the image to this size if size != []; len(size) == 2 required.
+
+  Returns:
+    np.ndarray: Contains object references to the chunks of the images (see load_chunk).
+  """
+
+  return [load_tarfile_from_s3(bucket, s3_key, size) for s3_key in s3_keys]
+
+def setup_variables(params, placeholders, assigns, kernelshape, biasshape):
+  """Creates the variables for each layer and adds the variables and the components needed to feed them to various lists
+
+  Args:
+    params (List): Network parameters used for creating feed_dicts
+    placeholders (List): Placeholders used for feeding weights into
+    assigns (List): Assignments used for actually setting variables
+    kernelshape (List): Shape of the kernel used for the conv layer
+    biasshape (List): Shape of the bias used
+
+  Returns:
+    None
+  """
+  kernel = tf.Variable(tf.zeros(kernelshape, dtype=tf.float32))
```

Comment:
Since now you aren't passing in the random weights from the driver, you'll have to change the initial values for the variables.